### ISSUE TITLE
Exporting geckodriver 0.29.1 fails for Firefox 91.9

### ISSUE THREAD

**Issue by slissner** [bonigarcia/webdrivermanager#832]
**Description of the problem**: Exporting geckodriver 0.29.1 (resolved driver for Firefox 91) fails because the `geckodriver` command is invoked with incompatible '--websocket-port' CLI argument.

It seems that the `--websocket-port` argument was added with geckodriver 0.30.0. Likely, we must use this driver version in order to fix the issue (see this [stackoverflow answer](https://stackoverflow.com/a/70033524/3757139)).

However, web driver manager uses version 0.29.x for Firefox 91:

```
firefox91=0.29.0
firefox91.linux=0.29.1
firefox91.win=0.29.1
```

**Browser and version**: Firefox 91.9

**Operating system**: Debian GNU/Linux 10 (buster)

**WebDriverManager version**: 5.1.1

**WebDriverManager call**: `WebDriverManager.firefoxdriver().setup();`

**WebDriverManager traces**: 

```
2022-05-23 10:24:28.602  INFO 7 --- [nio-8080-exec-3] i.g.bonigarcia.wdm.WebDriverManager      : Using geckodriver 0.29.1 (resolved driver for Firefox 91)
2022-05-23 10:24:28.612  INFO 7 --- [nio-8080-exec-3] i.g.bonigarcia.wdm.WebDriverManager      : Exporting webdriver.gecko.driver as /root/.cache/selenium/geckodriver/linux64/0.29.1/geckodriver
/root/.cache/selenium/geckodriver/linux64/0.29.1/geckodriver: error: Found argument '--websocket-port' which wasn't expected, or isn't valid in this context

USAGE:
    geckodriver --port <PORT>

For more information try --help

2022-05-23 10:24:28.680 DEBUG 7 --- [nio-8080-exec-3] o.s.web.servlet.DispatcherServlet        : Failed to complete request: org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
2022-05-23 10:24:28.684 ERROR 7 --- [nio-8080-exec-3] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.] with root cause

org.openqa.selenium.WebDriverException: Driver server process died prematurely.
Build info: version: '4.1.4', revision: '535d840ee2'
System info: host: '259e53f4859a', ip: '172.24.0.5', os.name: 'Linux', os.arch: 'amd64', os.version: '5.10.25-linuxkit', java.version: '11.0.7'
Driver info: driver.version: FirefoxDriver
	at org.openqa.selenium.remote.service.DriverService.start(DriverService.java:226) ~[selenium-remote-driver-4.1.4.jar!/:na]
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:98) ~[selenium-remote-driver-4.1.4.jar!/:na]
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:567) ~[selenium-remote-driver-4.1.4.jar!/:na]
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:248) ~[selenium-remote-driver-4.1.4.jar!/:na]
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:164) ~[selenium-remote-driver-4.1.4.jar!/:na]
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:118) ~[selenium-firefox-driver-4.1.4.jar!/:na]
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:106) ~[selenium-firefox-driver-4.1.4.jar!/:na]
```

**Comment by slissner** [comment]
Indeed, the [above mentioned](https://github.com/bonigarcia/webdrivermanager/issues/832#issue-1244946124) error message stems from selenium web driver and not from web driver _manager_: `/root/.cache/selenium/geckodriver/linux64/0.29.1/geckodriver: error: Found argument '--websocket-port' which wasn't expected, or isn't valid in this context`.

I think a breaking change was introduced in selenium firefox driver with commit https://github.com/SeleniumHQ/selenium/commit/728c746cd1573f79f5ec4a506b3ff84a9ac91279 ([selenium-4.0.0-rc-2](https://github.com/SeleniumHQ/selenium/releases/tag/selenium-4.0.0-rc-2))

For my java project, I upgraded the selenium dependencies form `v3.141.59` to `v4.1.4`. The dependencies are now:

```java
        <dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-java</artifactId>
            <version>4.1.4</version>
        </dependency>
        <dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-firefox-driver</artifactId>
            <version>4.1.4</version>
        </dependency>
```

So in summary, the incompatiblity stems from 

- web driver manager picking geckodriver v0.29 (for firefox 91.9), 
- while using most recent selenium-java packages (v4.1.4) that seems to require a newer geckodriver version (at least v0.30 I believe).

As this may not be an issue with webdrivermanager itself, I suggest to close this issue here.

However, maybe someone from the maintainers have any further suggestions how to avoid the issue. I am basically just using the newest versions of `selenium-java`, `selenium-firefox-driver` and `webdrivermanager` and they do not play well together.

**Comment by slissner** [comment]
Taking all of this into account: we now pin geckodriver to `v0.30.0` via environment
variable (`WDM_GECKODRIVERVERSION=0.30.0`) in order to workaround the problems.

**Comment by bonigarcia** [comment]
Thanks for reporting. I have changed the default mapping for Firefox 91. From now on, WebDriverManager will resolve geckodriver 0.30.0 for that version by default.

**Comment by slissner** [comment]
Thank you @bonigarcia! Looking forward for the next release :)

**Comment by bonigarcia** [comment]
You don't need to wait until the next release. The versions mapping is updated using the online info (in GitHub), so this information is already being used by WebDriverManager.

**Comment by slissner** [comment]
> You don't need to wait until the next release. The versions mapping is updated using the online info (in GitHub), so this information is already being used by WebDriverManager.

Ok, thank you for the information.

All fine now: I removed the environment variable (`WDM_GECKODRIVERVERSION=0.30.0`) and tested the running of the firefox driver. version `0.30.0` of geckodriver gets exported and it works like charm out of the box. 👍
